pygridtools/tests/test_iotools.py

```diff
@@ -98,3 +98,87 @@ def test_interactive_grid_shape(simple_grid):
     assert isinstance(widget.children[0], ipywidgets.IntSlider)
     assert isinstance(widget.children[1], ipywidgets.IntSlider)
     assert widget.children[0].max == widget.children[1].max == 100
+
+
+class Test__FocusProperties(object):
+    def __init__(self):
+        self.fp = iotools._FocusProperties()
```

Comment:
Do this as a fixture and break out the test class methods into their own functions

Reply:
Do you want a test function for each method?

Reply:
yeah. you'd parametrize it like this:

```python
@pytest.fixture
def fp():
    return iotools._FocusProperties(0.2, 'x', 0.3, 0.4)


@pytest.mark.parametrize(('attrname', 'attrtype'), [
    ('factor', float),
    ('axis', str),
    ('reach', float),
    ('extent', float)
])
def test_fp_attributes(fp, attrname, attrtype):
    attr = getattr(fp, attrname)  # effectively asserts hasattr
    assert isinstance(attr, attrtype)
```